examples/implementation/platform_independent/SDF_example_002/main_platform_independent.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include "common.h"
+
+/* Netlist */
+
+/*
+system :: Signal Int -> Signal Int -> Signal Int
+system s_ina s_inb = s_out where
+         s_1 = actor_a s_ina
+         s_2 = actor_b s_inb
+         s_3 = actor_c s_1 s_4_delayed
+(s_4, s_out) = actor_d s_2 s_3
+ s_4_delayed = delaySDF [0] s_4
+*/
+
+
+/* Process specification: */
+
+/*
+actor_a :: Signal Int -> Signal Int
+actor_a = actor11SDF 2 1 f_1 where
+    f_1 [x, y] = [x + y]
+
+actor_b :: Signal Int -> Signal Int
+actor_b = actor11SDF 1 2 f_2 where
+    f_2 [x] = [x, x+1]
+
+actor_c :: Signal Int -> Signal Int -> Signal Int
+actor_c = actor21SDF (2,1) 1 f_3 where
+    f_3 [x, y] [z] = [x + y + z]
+
+actor_d :: Signal Int -> Signal Int -> (Signal Int, Signal Int)
+actor_d = actor22SDF (2,1) (1,2) f_4 where
+    f_4 [x, y] [z] = ([x + y + z], [x + y, x + y + z])
+*/
+
+
+// A = actor11SDF 2 1 f_a where
+//  f_a [x, y] = [x + y]
+void f_a(token* in, token* out) {
```

Comment:
These ones still have the asterisk attached to the type rather than the variable.